packages/compass-telemetry/src/telemetry-events.ts

```diff
@@ -0,0 +1,1435 @@
+type ConnectionScoped<E extends { payload: unknown }> = E & {
+  payload: E['payload'] & { connection_id: string };
+};
+
+/**
+ * This event is fired when user successfully signed in to their Atlas account
+ *
+ * @category Atlas
+ */
+type AtlasSignInSuccessEvent = {
+  name: 'Atlas Sign In Success';
+  payload: { auid: string };
+};
+
+/**
+ * This event is fired when user failed to sign in to their Atlas account
+ *
+ * @category Atlas
+ */
+type AtlasSignInErrorEvent = {
+  name: 'Atlas Sign In Error';
+  payload: { error: string };
+};
+
+/**
+ * This event is fired when user signed out from their Atlas account
+ *
+ * @category Atlas
+ */
+type AtlasSignOutEvent = {
+  name: 'Atlas Sign Out';
+  payload: { auid: string };
+};
+
+/**
+ * This event is fired when user selects a use case from the aggregation panel
+ *
+ * @category Aggregation Builder
+ */
+type AggregationUseCaseAddedEvent = ConnectionScoped<{
+  name: 'Aggregation Use Case Added';
+  payload: { drag_and_drop?: boolean; stage_name?: string };
+}>;
+
+/**
+ * This event is fired when user adds/remove a stage or changes the stage name in the stage editor view
+ *
+ * @category Aggregation Builder
+ */
+type AggregationEditedEvent = ConnectionScoped<{
+  name: 'Aggregation Edited';
+  payload: {
+    num_stages?: any | number;
+    editor_view_type?: string | 'stage' | 'text' | 'focus';
```

Comment:
so just string?

Reply:
similarly if num_stages is any | number it might as well be any?

Reply:
These types are extracted with a script, that's why i got `string` in some places. In my opinion, where the values are meaningful for analysis we should keep the stricter type and document them, my intention is to address any manual tweak in a follow up:

> TODO (in a follow up so new events aren't added in the meanwhile):
>
> use stricter, more precise types for properties
> document properties
> generate tracking plan in markdown